### PR TITLE
znc: Bumped to v1.6.3, add ICU support and user configuration.

### PR DESCRIPTION
**Pull request by Xotic750**
Bumped to v1.6.3
Added support for ICU through menu configuration.
Added RUNAS_SHELL variable to assist when using external configuration.
Signed-off-by: Graham Fairweather <xotic750@gmail.com>

### CODE REVIEW

File: net/znc/Config.in [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+config ZNC_ICU
+  bool
+  prompt "Enable ICU character encoding support."
+  help
+    ZNC has supported character encodings in core since 1.6.0. If you want to
+    look at the earlier "charset" module, it's located at Charset/old. ZNC uses
+    UTF-8 internally. You can configure charset used between clients and ZNC,
+    and charset used between ZNC and the IRC server. Likely values are UTF-8
+    and ISO-8859-1, but you might need other common character encodings.
```
Comment (KanjiMonster) [inline net/znc/Config.in]: You don't need include the history of this option here, please reduce it to one or two sentences. A short description why you would want that is enough.

File: net/znc/Config.in [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  default n
```
Comment (KanjiMonster) [inline net/znc/Config.in]: n is already default, so just remove this line.

File: net/znc/Makefile [hunk 3]
```diff
@@ -23,6 +23,8 @@ PKG_LICENSE_FILES:=LICENSE
 PKG_USE_MIPS16:=0
 PKG_BUILD_PARALLEL:=1
 
+PKG_CONFIG_DEPENDS := \
+	CONFIG_ZNC_ICU
```
Comment (KanjiMonster) [inline net/znc/Makefile]: No need to split this into two lines.

File: net/znc/Makefile [hunk 4]
```diff
@@ -290,6 +295,12 @@ CONFIGURE_ARGS += \
 	--enable-tdns \
 	--enable-zlib
 
+ifeq ($(CONFIG_ZNC_ICU),y)
+	CONFIGURE_ARGS += --enable-charset
+else
+	CONFIGURE_ARGS += --disable-charset
+endif
+
```
Comment (KanjiMonster) [inline net/znc/Makefile]: Just replace the removed line with
`$(if $(CONFIG_ZNC_ICU), --enable-charset, --disable-charset))`

File: net/znc/files/znc.init [hunk 5]
```diff
@@ -168,15 +170,21 @@ start() {
...
 		if [ "$SU" ]
 		then
-			$SU -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER
+			chown -hR ${RUNAS_USER:-nobody}:${RUNAS_GROUP:-nogroup} ${ZNC_CONFIG_PATH}
+			if [ "$RUNAS_SHELL" ]
+			then
+				$SU -s $RUNAS_SHELL -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER
+			else
+				$SU -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER
+			fi
```
Comment (KanjiMonster) [inline net/znc/files/znc.init]: You can write this shorter as 
`$SU ${RUNAS_SHELL:+s $RUNAS_SHELL} -c "/usr/bin/znc -d$ZNC_CONFIG_PATH >/dev/null &" $RUNAS_USER`

**Comment by KanjiMonster** [comment]
You are doing three things at once, please split this into three commits:
1) Bumped to v1.6.3
2) Added support for ICU through menu configuration.
3) Added RUNAS_SHELL variable to assist when using external configuration.

Also please sign off the commits them selves. And the correct format is 
"Signed-off-by: Name <emailaddress>"

**Comment by KanjiMonster** [comment]
Also please use the prefix "znc: " for the commit subjects

File: net/znc/Config.in (hunk 1 above)
Comment (Xotic750) [inline net/znc/Config.in]: "Allows configuration of the charset used between clients and ZNC, and between ZNC and the IRC server"

Better?

File: net/znc/Config.in (hunk 2 above)
Comment (Xotic750) [inline net/znc/Config.in]: As this is my first delve into a package, I have been verbose as I am unsure of what is necessary or not. Noted.

File: net/znc/Makefile (hunk 3 above)
Comment (Xotic750) [inline net/znc/Makefile]: I originally had an option for SSL too, as all other packages seem to disable it by default but I took it out. Noted.

File: net/znc/Makefile (hunk 4 above)
Comment (Xotic750) [inline net/znc/Makefile]: Ok cool. Again verbose to what I understood.

File: net/znc/files/znc.init (hunk 5 above)
Comment (Xotic750) [inline net/znc/files/znc.init]: Again. :)

**Comment by Xotic750** [comment]
Ok, will split it into 3.

I don't see what the problem is with the sign off, I am using exactly what is stated in the documents it is just that I opted out of displaying my email address in the options, so you can't see it in this post but it is actually in the message as I can see it if I choose to edit it?

znc: noted.

**Comment by Xotic750** [comment]
1) https://github.com/openwrt/packages/pull/2785

**Pull request closed without merging**